src/services/supabase.ts

```diff
@@ -107,3 +111,88 @@ export const getUserDetails = (user: User | null) => {
         avatar,
     };
 };
+
+export interface CallSupabaseResponse {
+    error?: PostgrestError;
+    data: any;
+}
+
+const handleSuccess = (response: any) => {
+    if (response.error) {
+        return {
+            data: null,
+            error: response.error,
+        };
+    } else {
+        return {
+            data: response.data,
+        };
+    }
+};
+
+const handleFailure = (error: any) => {
+    return {
+        data: null,
+        error,
+    };
+};
+
+export const insertSupabase = (
+    table: TABLES,
+    data: any
+): PromiseLike<CallSupabaseResponse> => {
+    const query = supabaseClient.from(table);
+
+    const makeCall = () => {
+        return query.insert([data]).then(handleSuccess, handleFailure);
+    };
+
+    return makeCall();
+};
+
+// Used to make update calls. Mainly consumed in the src/api folder
+export const supabaseUpdate = (
```

Comment:
This function name should begin with a verb to align with the naming convention: `updateSupabase`.